build-tools/tasks/test-utils.js

```diff
@@ -18,12 +19,59 @@ const configs = {
   dom: {
     defaultExport: `export default function wrapper(root: Element = document.body) { if (document && document.body && !document.body.contains(root)) { console.warn('[AwsUi] [test-utils] provided element is not part of the document body, interactions may work incorrectly')}; return new ElementWrapper(root); }`,
     buildFinderInterface: ({ componentName }) =>
-      `find${componentName}(selector?: string): ${toWrapper(componentName)} | null;`,
+      `/**
+        * Returns the wrapper of the first ${componentName} that matches the specified CSS selector.
+        * If no CSS selector specified, returns the wrapper of the first ${componentName}.
+        * If no matching ${componentName} is found, returns \`null\`.
+        *
+        * @param {string} [selector] CSS Selector
+        * @returns {${toWrapper(componentName)} | null}
+        */
+       find${componentName}(selector?: string): ${toWrapper(componentName)} | null;
+       /**
+        * Returns the wrappers of all ${pluralize(componentName)} that match the specified CSS selector.
+        * If no CSS selector specified, returns all of the ${pluralize(componentName)} inside current wrapper.
+        * If no matching ${componentName} is found, returns an empty array.
+        *
+        * @returns {Array<${toWrapper(componentName)}>}
+        */
+       findAll${pluralize(componentName)}(selector?: string): Array<${toWrapper(componentName)}>;
+       /**
+        * Returns the first ${componentName} that matches the specified test id.
+        * Looks for \`data-testid\` attribute assigned to the component.
+        * If no matching ${componentName} is found, returns \`null\`.
+        *
+        * @param {string} testId
+        * @returns {${toWrapper(componentName)} | null}
+        */
+       find${componentName}ByTestId(testId: string): ${toWrapper(componentName)} | null;`,
   },
   selectors: {
     defaultExport: `export default function wrapper(root: string = 'body') { return new ElementWrapper(root); }`,
     buildFinderInterface: ({ componentName }) =>
-      `find${componentName}(selector?: string): ${toWrapper(componentName)};`,
+      `/**
+        * Returns a wrapper that matches a ${componentName} with the specified CSS selector.
+        * If no CSS selector is specified, returns a wrapper that matches ${componentName}.
```

Comment:
For component names starting with a vowel, the resulting text could be suboptimal since we don't use "an", e.g. it would produce "that matches a Alert" instead of "that matches an Alert". I would recommend that we align the phrasing here with the one used for the unit tests selectors, which resolves this issue too:

```suggestion
        * Returns a wrapper that matches the first ${componentName} with the specified CSS selector.
        * If no CSS selector is specified, returns a wrapper that matches the first ${componentName}.
```